feat!: add templates for commonly used objects

Wielewout opened the pull request:
This PR revamps the common library charts by adding templates for commonly used k8s objects (like deployment) within the 5g charts (cu-cp, cu-up and cell-wrapper).

The charts are renamed to just `common` as they should already be part of the `accelleran` repository in helm.

Wielewout committed:
feat: add templates for common k8s objects

feat: add init container checks

feat: add bootstrap configmap template

feat!: rename chart to common

feat!: use prefix accelleran.common for templates

feat!: merge instance id checks into main template

Wielewout committed:
fix: fix drax role and technology

Wielewout committed:
fix: allow to call pod tpl without container values

Wielewout committed:
feat: add service support for loadbalancer IP and external traffic policy

Wielewout committed:
fix: use metadata tpl accross other templates

Wielewout committed:
feat: add host network and dns policy pod config

Wielewout committed:
feat: add init container tpl

Wielewout committed:
fix: add deprecated labels

Wielewout committed:
fix: add part-of label

Wielewout committed:
feat: add appProtocol to service tpl

Wielewout committed:
fix: fix empty label values

Wielewout committed:
feat: allow to set replica count to 0 (also when autoscaling)

Wielewout committed:
docs: add readme

Wielewout committed:
refactor: rename to annotations

Wielewout committed:
refactor: rename to labels

Wielewout committed:
refactor!: use global nodeselector instead of drax nodeselector

Wielewout committed:
fix!: separate init image definitions for nats and redis

Wielewout committed:
feat: allow to set node name in a pod

Wielewout committed:
feat: add fallback in case nats and redis are installed as part of the chart

This assumes that the official nats service is used
as well as the redis chart by bitnami.

Wielewout committed:
feat: allow to set image pull secret in global

Wielewout committed:
fix!: move bootstrap from global to top level of values

Also removes attempts for backwards compatibility.

Wielewout committed:
fix!: move drax from global to top level of values

Wielewout committed:
chore!: remove deprecated labels

Wielewout committed:
fix: use /bin/sh instead of bash

Wielewout committed:
feat: add kafka to init containers and bootstrap config

Wielewout committed:
feat: add flag to explicitly  indicate what bootstrap config should be added

Wielewout committed:
feat: add ports to container

refactor: move service ports in template

Wielewout committed:
feat: add cluster rbac

Wielewout committed:
fix: prevent error with undefined image

Wielewout committed:
feat: add drax license

ben-foxmoore reviewed (approved):
LGTM, apart from a couple of documentation questions.

Wielewout committed:
feat!: allow multiple ports in a service and container

Wielewout committed:
docs: clarify and fix language

Wielewout committed:
feat!: use v1 cron jobs

Wielewout committed:
feat: allow to set custom selector labels

Wielewout committed:
fix: switch to kafka provisioned topic for check-kafka init containers

Wielewout committed:
fix: fix error when values global not set properly

Wielewout committed:
fix: add timeout of 10s to check-redis init containers

Wielewout committed:
feat: add tpl to bootstrap config to allow chart reuse

Wielewout committed:
feat!: remove instance filter from bootstrap configmap

As the instance filter needs to be different for each deployment of cw, cucp, cuup,
...
it is extracted to allow to re-use a global configmap for just the
connection details.

Wielewout committed:
fix: prevent warning of duplicate port definition
